internal/builders/bazel/rebuilder.sh

```diff
@@ -0,0 +1,490 @@
+
+# Obtain the name of the artifact
+if [[ $artifact_path == */* ]]
+then
+    artifact_name=$(basename "$artifact_path")
+else
+    artifact_name=$artifact_path
+fi
+
+rebuilt_checksum=""
+unset rebuilt_checksum # Makes sure it is empty before assigning.
+
+# IF there are runfiles, the directory structure will be different.
+# The binaries folder contains different directories for the its artifacts and
+# the artifacts runfiles. Obtain the rebuilt binaries and copy them to the
+# path at root before cleaning up and deleting the repo.
+if [[ "$artifact_name" == *"_deploy.jar"* ]]
+then
+      # Uses _deploy.jar as a field seperator and grabs the field before it.
+      # Directory of Java artifacts is same as run script name.
+      run_script_name=$(echo "$artifact_name" | awk -F'_deploy.jar' '{print $1}')
+      cd $binaries_dir/
+      rebuilt_checksum=$(sha256sum ./"$run_script_name"/"$artifact_name" | awk '{ print $1 }')
+
+      # Copy the entire directory, including the run script.
+      cp -R ./"$run_script_name" ./../../"$rebuilt_artifacts_dir"/
+else
+    if [[ "${NEEDS_RUNFILES}" == "true" ]]
+    then
+        # For non-java targets with runfiles.
+      cd $binaries_dir/
+      rebuilt_checksum=$(sha256sum ./"$artifact_name"/"$artifact_name" | awk '{ print $1 }')
+
+      # Copy entire directory, including the runfiles.
+      cp -R ./"$artifact_name" ./../../"$rebuilt_artifacts_dir"/
+    else
+    # For files withouts runfiles.
+    cd $binaries_dir
+    rebuilt_checksum=$(sha256sum "$artifact_name" | awk '{ print $1 }')
+
+    cp "$artifact_name" ./../../$rebuilt_artifacts_dir/
+    fi
+fi
+
+################################################
+#                                              #
+#       Check Build for Reproducibility        #
+#                                              #
+################################################
+
+if [[ "$orig_checksum" == "$rebuilt_checksum" ]]
+then
+    printf "${GREEN}Checksum is the ${BOLD}${UNDERLINE}same${RESET}${GREEN} for the original and rebuilt artifact!${RESET}\n"
+    printf "${GREEN}✅ This build is ${BOLD}${UNDERLINE}reproducible!${RESET} ✅ \n"
+    echo ""
+    printf "${GREEN}%s${RESET} = Original Checksum${RESET}\n" "$orig_checksum"
+    printf "${GREEN}%s${RESET} = Rebuilt Checksum${RESET}\n" "$rebuilt_checksum"
+    echo ""
+else
+    printf "${BOLD_RED_BG}Checksum is ${BOLD}${UNDERLINE}NOT${RESET}${BOLD_RED_BG} the same for the original and rebuilt artifact!${RESET}\n"
+    printf "${BOLD_RED_BG}        ⚠️  This build was ${BOLD}${UNDERLINE}NOT${RESET}${BOLD_RED_BG} able to be reproduced! ⚠️         ${RESET}\n"
+    echo ""
+    printf "${RED}%s${RESET} = Original Checksum\n" "$orig_checksum"
+    printf "${RED}%s${RESET} = Rebuilt Checksum\n" "$rebuilt_checksum"
+    echo ""
+fi
+
+
+if [[ cleanup -eq 1 ]]
+then
+
+  # If there are runfiles or if the artifacts are Java, then each artifact
+  # has its own directory, so you need to exit out of it first.
```

Comment:
This code block should also be indented consistently with the rest of the script (4 spaces)

Reply:
indented!